Tests/SwerverTests/RouterSpec/RouterSpec.swift

```diff
@@ -9,53 +9,101 @@ class RouterSpec: QuickSpec {
+                        let aRogueRequest = try httpRequestParser.parse(request: requestString)
+                        
+                        let expectedResponse = responseBuilder
+                                                    .withStatusCode(statusCode: StatusCode.not_found.rawValue)
+                                                    .withStatusPhrase(statusPhrase: StatusCode.not_found.getStatusPhrase())
+                                                    .withContentType(contentType: ContentType.text.rawValue)
+                                                    .build()
+                        expect(router.process(request: aRogueRequest)).to(equal(formatter.format(httpResponse: expectedResponse)))
+                    } catch {}
                 }
             }
 
             context("when receive an OPTIONS request to an existing resource") {
-                it("returns an response with all the allowed methods of that resource") {
-                    let anOptionsRequest = HttpRequest(method: RequestMethod.options, url: URL(string: Resource.test.rawValue)!, httpVersion: HttpVersion.current)
-                    let expectedResponse = responseBuilder
-                        .withStatusCode(statusCode: StatusCode.ok.rawValue)
-                        .withStatusPhrase(statusPhrase: StatusCode.ok.getStatusPhrase())
-                        .withContentType(contentType: ContentType.text.rawValue)
-                        .withAllowedMethods(allowedMethods: ["HEAD", "GET"])
-                        .build()
-                    expect(router.process(request: anOptionsRequest)).to(equal(formatter.format(httpResponse: expectedResponse)))
+                it("returns a 200 OK response with all the allowed methods of that resource") {
+                    
+                    do {
+                        let requestString = "\(RequestMethod.options.rawValue) \(Resource.test.rawValue) \(HttpVersion.current.rawValue)"
+                        
+                        let anOptionsRequest = try httpRequestParser.parse(request: requestString)
+                        
+                        let expectedResponse = responseBuilder
+                                                    .withStatusCode(statusCode: StatusCode.ok.rawValue)
+                                                    .withStatusPhrase(statusPhrase: StatusCode.ok.getStatusPhrase())
+                                                    .withContentType(contentType: ContentType.text.rawValue)
+                                                    .withAllowedMethods(allowedMethods: ["HEAD", "GET"])
+                                                    .build()
+                        
+                        expect(router.process(request: anOptionsRequest)).to(equal(formatter.format(httpResponse: expectedResponse)))
+                    } catch {}
+                }
+            }
+
+            context("when receive a unknown request to an existing resource") {
```

Comment:
Try-catch blocks within a test are a sign that you might be able to add in a new level of abstraction (note that this is different than testing something throws an exception, which is fine). Is there some refactoring that you can do to your code to clean these tests up?

Reply:
will look into that

Reply:
addressed in the last commit.

Reply:
Having `parse` return an `HttpRequest?` instead of using throw inside of `parse` might help. This is something I thought of long after we discussed using throw.

Reply:
fixed in feature-remove-forced-unwrapping branch.